OpenAPITools/openapi-generator
# Adds Collapse Spec Options to Maven Plugin

* Adds a collapsedSpec option to the maven plugin that produces a single-file representation of the spec in the output directory.
* Adds a includeCollapsedSpecInArtifacts option to the maven plugin that adds the collapsed spec file to the maven artifacts.

### How to manually verify
1. Add the maven plugin to a project.
2. Add the `collapsedSpec` and `includeCollapsedSpecInArtifacts` options to the plugin configuration.
3. Run `mvn deploy`
4. Verify that the target directory contains the merged spec. This file name will match the `collapsedSpec` value.
5. Verify the deploy repository contains the spec where the classifier name matches that of the `collapsedSpec` value.




### PR checklist
 
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
